docs/rules/no-restricted-properties.md

```diff
@@ -0,0 +1,73 @@
+# Disallow certain object properties (no-restricted-properties)
+
+Certain properties on objects may be disallowed in a codebase. This is useful for deprecating an API or restricting usage of a module's methods. This rule looks for accessing a given property key on a given object name, either when reading the property's value or invoking it as a function. You may specify an optional message to indicate an alternative API or a reason for the restriction.
+
+## Rule Details
+
+This rule is aimed at disallowing certain object properties from your code.
+
+### Options
+
+This rule takes a list of objects, where the object name and property names are specified:
+
+```json
+{
+    "rules": {
+        "no-restricted-properties": [2, [{
+            "object": "disallowedObjectName",
+            "property": "disallowedPropertyName"
+        }]]
+    }
+}
+```
+
+Multiple object/property values can be disallowed, and you can specify an optional message:
+
+```json
+{
+    "rules": {
+        "no-restricted-properties": [2, [{
```

Comment:
This might be the wrong time to bring this up, but could the option syntax be simplified like so:

```js
// ...
  "no-restricted-properties": [ "error", [ {
    "objectName": {
      "propertyName1": "Error message 1",
      "propertyName2": "Error message 2"
    },
    "anotherObjectName": {
       // ...
    }, // etc
  } ] ],
```